Net6APIBasicAuthApiKey/Auth/AuthorizationService.cs

```diff
@@ -17,34 +17,32 @@ public AuthorizationService(ILogger<AuthorizationService> logger, IOptions<Servi
         }
         public Task<AuthorizationResult> AuthorizeAsync(ClaimsPrincipal user, object? resource, IEnumerable<IAuthorizationRequirement> requirements)
         {
-            if (resource is HttpContext httpContext && user.Identity is not null)
+            if (resource is not HttpContext httpContext || user.Identity is null) 
             {
-                var ip = httpContext.Connection.RemoteIpAddress;
+                return Task.FromResult(AuthorizationResult.Failed(AuthorizationFailure.Failed(requirements)));
+            }
 
-                if (!user.Identity.IsAuthenticated)
-                {
-                    _logger.LogInformation($"Unauthorized access {ip}");
-                    return Task.FromResult(AuthorizationResult.Failed(AuthorizationFailure.Failed(requirements)));
-                }
+            var ip = httpContext.Connection.RemoteIpAddress;
 
-                if (!httpContext.Request.Headers.ContainsKey(ApiKeyHeaderValue))
-                {
-                    _logger.LogInformation($"Missing API-KEY:{ip}");
-                    return Task.FromResult(AuthorizationResult.Failed(AuthorizationFailure.Failed(requirements)));
-                }
+            if (!user.Identity.IsAuthenticated)
+            {
+                _logger.LogInformation("Unauthorized access {}", ip);
+                return Task.FromResult(AuthorizationResult.Failed(AuthorizationFailure.Failed(requirements)));
+            }
 
-                var headerApiKey = httpContext.Request.Headers[ApiKeyHeaderValue];
-                if (_serviceAccessInfo.ApiKey != headerApiKey)
-                {
-                    _logger.LogInformation($"Invalid API-KEY:{headerApiKey}:{ip}");
-                    return Task.FromResult(AuthorizationResult.Failed(AuthorizationFailure.Failed(requirements)));
-                }
-                return Task.FromResult(AuthorizationResult.Success());
+            if (!httpContext.Request.Headers.ContainsKey(ApiKeyHeaderValue))
+            {
+                _logger.LogInformation("Missing API-KEY:{}", ip);
+                return Task.FromResult(AuthorizationResult.Failed(AuthorizationFailure.Failed(requirements)));
             }
-            else
+
+            var headerApiKey = httpContext.Request.Headers[ApiKeyHeaderValue];
+            if (_serviceAccessInfo.ApiKey != headerApiKey)
             {
+                _logger.LogInformation("Invalid API-KEY:{}:{}", headerApiKey, ip);

```

Comment:
Format string power.. I like the {0} {1} ... or {ip} style.. much readable when you use multiple params..
The empty brackets is avoidable for me

Reply:
I think that the {ip} style is the more descriptive.